pbj-core/pbj-runtime/src/test/java/com/hedera/pbj/runtime/ProtoWriterToolsTest.java

```diff
@@ -142,6 +152,16 @@ void testWriteTagSpecialWireType() {
         assertEquals((definition.number() << TAG_TYPE_BITS) | WIRE_TYPE_FIXED_64_BIT.ordinal(), bufferedData.readVarInt(false));
     }
 
+    @Test
+    void testWriteInteger_zero() {
+        FieldDefinition definition = createFieldDefinition(INT32);
+        final int valToWrite = 0;
+        final long positionBefore = bufferedData.position();
+        writeInteger(bufferedData, definition, valToWrite);
+        final long positionAfter = bufferedData.position();
+        assertEquals(positionBefore, positionAfter);
+    }
+
     @Test
     void testWriteInteger_int32() {
         FieldDefinition definition = createFieldDefinition(INT32);
```

Comment:
This test case writes a random int. If the int happens to be 0 (the default int value), the test would fail. Does it make sense to protect from this as a part of current PR?

Reply:
Good catch. Addressed it for this and a few more subsequent tests.